AngularJS/src/org/angular2/entities/source/Angular2SourceDirective.java

```diff
@@ -143,6 +161,51 @@ public boolean isTemplate() {
                 Collections.unmodifiableCollection(outputs.values()));
   }
 
+  @NotNull
+  private Angular2CtorBoundParameters getCtorParameters() {
+    // The @Attribute decorated parameters can be found (and are valid) only
+    // in the class constructor, thus we have to get hold of it and process each parameter
+    final JSFunction ctor = getTypeScriptClass().getConstructor();
+    return Objects.isNull(ctor)
+          ? Angular2CtorBoundParameters.empty()
+          : processCtorParameters(ctor);
+  }
+
+  /**
+   * Analyze the class constructor's parameters to find
+   * {@code @Attribute} bound ones.
+   *
+   * @param ctor
+   * 			The class constructor
+   */
+  @NotNull
+  private Angular2CtorBoundParameters processCtorParameters(final JSFunction ctor) {
+    final Map<String, Angular2DirectiveCtorParameter> attributes = new LinkedHashMap<>(8);
+
+    for (final JSParameter ctorParam : ctor.getParameterVariables()) {
+      final JSAttributeList attributeList = ctorParam.getAttributeList();
+
+      if (Objects.isNull(attributeList)) {
+        continue;
+      }
+
+      final String paramName = ctorParam.getName();
+      final Optional<String> bindingName =
+            Arrays.stream(attributeList.getDecorators())
+                  .filter(d -> Angular2DecoratorUtil.ATTRIBUTE_DEC.equals(d.getDecoratorName()))
+                  .findFirst()
+                  .map(Angular2SourceDirective::getStringParamValue)
+                  .map(n -> StringUtil.notNullize(n, paramName));
```

Comment:
Instead of having if below you could:
`.ifPresent(key -> attributes.putIfAbsent(key, new Angular2SourceDirectiveDecoratedCtorParameter(key, ctorParam))`